### ISSUE TITLE
kerasR

### ISSUE THREAD

**Issue by statsmaths** [ropensci/onboarding#117]
### Summary

-  What does this package do? (explain in 50 words or less):

```
Provides a consistent interface to the Keras Deep Learning Library directly from within R.
Keras is a library for describing neural networks (Dense, Convolutional, and Recurrent),
training them on data, and applying to new datasets. Several popular pre-trained models
are also included in the package.
```

-   Paste the full DESCRIPTION file inside a code block below:

```
Provides a consistent interface to the Keras Deep Learning Library
directly from within R. Keras provides specifications for describing dense
neural networks, convolution neural networks (CNN) and recurrent neural networks
(RNN) running on top of either TensorFlow or Theano. Type conversions between
Python and R are automatically handled correctly, even when the default
choices would otherwise lead to errors. Includes complete R documentation
and many working examples.
```

-   URL for the package (the development repository, not a stylized html page): [https://github.com/statsmaths/kerasR](https://github.com/statsmaths/kerasR)

- Please indicate which category or categories from our [package fit policies](https://github.com/ropensci/onboarding/blob/master/policies.md#package-fit) this package falls under and why? (e.g., data retrieval, reproducibility. If you are unsure, we suggest you make a pre-submission inquiry.): text-mining, image-processing

-   Who is the target audience? Anyone working with text and image data who wants to apply popular neural networks to their datasets. These can be used as a form of dimensionality reduction for visualization or exploratory purposes. Users can also load popular models (VGG-19, ResNet-15, InceptionV3) in order to reproduce research in other papers directly in R.

-   Are there other R packages that accomplish the same thing? If so, what is different about yours? There is a tensorflow package offering a low-level wrapper to the tensorflow library, but this does not allow users to fit and run neural networks without a lot of additional work and knowledge.

### Requirements

Confirm each of the following by checking the box.  This package:

- [X] does not violate the Terms of Service of any service it interacts with.
- [X] has a CRAN and OSI accepted license.
- [X] contains a README with instructions for installing the development version.
- [X] includes documentation with examples for all functions.
- [X] contains a vignette with examples of its essential functions and uses.
- [X] has a test suite.
- [X] has continuous integration, including reporting of test coverage, using services such as Travis CI, Coeveralls and/or CodeCov.
- [X] I agree to abide by [ROpenSci's Code of Conduct](https://github.com/ropensci/onboarding/blob/master/policies.md#code-of-conduct) during the review process and in maintaining my package should it be accepted.

#### Publication options

- [X] Do you intend for this package to go on CRAN?
- [X] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:
    - [X] The package contains a [`paper.md`](http://joss.theoj.org/about#paper_structure) with a high-level description in the package root or in `inst/`.
    - [X] The package is deposited in a long-term repository with the DOI: [https://CRAN.R-project.org/package=cleanNLP](https://CRAN.R-project.org/package=cleanNLP)
    - (*Do not submit your package separately to JOSS*)

### Detail

- [X] Does `R CMD check` (or `devtools::check()`) succeed?  Paste and describe any errors or warnings:

- [X] Does the package conform to [rOpenSci packaging guidelines](https://github.com/ropensci/packaging_guide)? Please describe any exceptions:

Names of classes in the keras library (such as Dense and Dropout) do not following the naming conventions in order to mirror the naming conventions in the keras library. This makes it easiest to port code between Python and R.

- If this is a resubmission following rejection, please explain the change in circumstances:

- If possible, please provide recommendations of reviewers - those with experience with similar packages and/or likely users of your package - and their GitHub user names: Mike Kane (@kaneplusplus), Andrie de Vries (@andrie), Akitaka Matsuo (@amatsuo), Adam Obeng (@adamobeng)

**Comment by kaneplusplus** [comment]
I am willing to review this.

**Comment by maelle** [comment]
Thanks for your submission @statsmaths! Unfortunately, after discussion we think that it is out of scope despite the applications you mention: it is a very broad standalone machine learning/statistical package. Let us know if you have any question, and in any case please know that we really appreciate your submitting your package! We encourage you to submit it to [JOSS](http://joss.theoj.org/).

Furthermore, for your information there seems to be another R interface to Keras: https://rstudio.github.io/keras/

@kaneplusplus thanks for volunteering, maybe you can volunteer to review this package for JOSS if @statsmaths chooses to go that way, but otherwise don't worry, we won't ignore your willingess to review a package and will ask you to review one sooner or later 😉

**Comment by statsmaths** [comment]
@maelle Thanks for the quick reply. Reading the on-boarding guidelines suggests that packages that are *just* broad standalone tools are out of scope, not that packages are immediately disqualified for being broad or doing machine learning. Yes, kerasR can do general purpose machine learning, but I was submitting it based on its application to text and image processing (arguably the most important applications of Neural Networks). 

Re-reading the onboarding guidelines I am not particularly clear one what "in-scope" means for rOpenSci. Packages like **tokenizers** and **wordVectors** don't seem fit into the narrow categories in the "Package fit" section either but are obviously important for scientific applications with text processing. I thought **kerasR** fit the same way. I was asked to review a data access package for grabbing crypto-currency data, which is a data retrieval project but I'm not sure how it passes the "Packages that fall into above categories but without any scientific applications" test.

I don't mean any this overly argumentatively or aggressively. Clearly the project has some idea of what they want to be considered "in-scope" and that's fine with me. I just wanted to point at that to me it does not seem particularly clear or consistent from what is posted as guidelines on the website. 

Thanks for the link the to rstudio project. I did not know about this and it seems that they have only released their repo publicly a few days ago. Mine has been on CRAN since March. It is quite frustrating that they did not even bother to contact me (would have have happily helped out with their efforts rather than duplicating effort).
